src/Encamina.Enmarcha.SemanticKernel.Abstractions/ILengthFunctions.cs

```diff
@@ -5,18 +5,54 @@ namespace Encamina.Enmarcha.SemanticKernel.Abstractions;
 /// <inheritdoc/>
 public interface ILengthFunctions : AI.Abstractions.ILengthFunctions
 {
+    /// <summary>
+    /// Dictionary to cache GptEncoding instances based on encoding names.
+    /// </summary>
+    private static readonly Dictionary<string, GptEncoding> EncodingCache = [];
+
+    /// <summary>
+    /// Gets the default GptEncoding for models like `GPT-3.5-Turbo` and `GPT-4` from OpenAI.
+    /// </summary>
+    private static GptEncoding defaultGptEncoding;
```

Comment:
I would suggest to convert this into a public read only and static member that is initialized with a call to `GptEncoding.GetEncoding("cl100k_base")` instead of a private variable and an additional method.

So, the code would be like:

>     ...
>     /// <summary>
>     /// Gets the default <see cref="GptEncoding">encoding</see> for models like `GPT-3.5-Turbo` and `GPT-4` from OpenAI.
>     /// </summary>
>     public static readonly GptEncoding DefaultGptEncoding = GptEncoding.GetEncoding("cl100k_base");
>     ...

And removed the method `GetDefaultGptEncoding`.